### PR TITLE
fix: completion invoke when next line has an indented comment

### PR DESCRIPTION
**Pull request by rickcowan**
### What does this PR do?
fix: completion invoke when next line has an indented comment

The change for this was to evaluate `range[1]` (value-end) rather than `range[2]` (node-end, which may contain trivia such as comments) when trying to find the closest node.

```yaml
example:
  prop1: "test"
  # cursor here
    #comment
```

### What issues does this PR fix or reference?
no ref

### Is it tested? How?
added unit tests

### CODE REVIEW

File: test/autoCompletionFix.test.ts [hunk 1]
```diff
@@ -538,4 +538,47 @@ objB:
...
+      const completion = await parseSetup(content, 2, 2);
+      expect(completion.items.length).equal(1);
+      expect(completion.items[0]).to.be.deep.equal(
+        createExpectedCompletion('prop2', 'prop2: ', 2, 2, 2, 2, CompletionItemKind.Property, InsertTextFormat.Snippet, {
+          documentation: ''
+        })
+      );
+    });
+
+    it('completion should handle comment at same indent level on new line', async () => {
```
Comment (rickcowan) [inline test/autoCompletionFix.test.ts:572]: This unit test is testing functionality that was already working before the fix. Simply added as a control check.

**Comment by gorkem** [comment]
Looks like it failed prettier checks. `yarn build` should have caught them. Can you update the PR with the fixes.

File: src/languageservice/parser/yaml-documents.ts [hunk 2]
```diff
@@ -137,7 +137,7 @@ export class SingleYAMLDocument extends JSONDocument {
       if (!range) {
         return;
       }
-      const diff = range[2] - offset;
+      const diff = range[1] - offset;
```
Comment (msivasubramaniaan) [inline src/languageservice/parser/yaml-documents.ts:140]: what could happened if next line hasn't comment?

Comment (rickcowan) [reply src/languageservice/parser/yaml-documents.ts:140]: Tested this manually and it works. Added another unit test for this situation.

**Comment by gorkem** [comment]
@p-spacek do you mind reviewing this one?

**Review by gorkem — APPROVED** [review]
LGTM

**Comment by gorkem** [comment]
Great catch, thank you for the contribution. I think this got a conflict on the test after another contribution was merged.

**Comment by p-spacek** [comment]
> @p-spacek do you mind reviewing this one?

We were cooperating together with Rick on this issue. So from my point of view, it's ok.
@rickcowan can you please check the conflict and tests after the merge?

**Pull request merged**